src/script/components/TitleBar/index.tsx

```diff
@@ -0,0 +1,226 @@
+import {matchQualifiedIds} from 'Util/QualifiedId';
+import {CallActions} from '../../view_model/CallingViewModel';
+import {handleKeyDown} from 'Util/KeyboardUtil';
+
+export interface TitleBarProps {
+  conversationEntity: Conversation;
+  legalHoldModal: LegalHoldModalViewModel;
+  showConversationDetails: (addParticipants: boolean) => void;
+  callingRepository: CallingRepository;
+  callActions: CallActions;
+  isPanelVisible: boolean;
+  userState: UserState;
+  callState: CallState;
+  teamState: TeamState;
+}
+
+const TitleBar: React.FC<TitleBarProps> = ({
+  conversationEntity,
+  legalHoldModal,
+  showConversationDetails,
+  callingRepository,
+  callActions,
+  isPanelVisible,
+  userState = container.resolve(UserState),
+  callState = container.resolve(CallState),
+  teamState = container.resolve(TeamState),
+}) => {
+  const badgeLabelCopy = useMemo(() => {
+    if (conversationEntity.is1to1() && conversationEntity.isRequest()) {
+      return '';
+    }
+
+    const hasExternal = conversationEntity.hasExternal();
+    const hasGuest = conversationEntity.hasDirectGuest();
+    const hasService = conversationEntity.hasService();
+    const hasFederated = conversationEntity.hasFederatedUsers();
+    const translationKey = generateWarningBadgeKey({hasExternal, hasFederated, hasGuest, hasService});
+
+    if (translationKey) {
+      return t(translationKey);
+    }
+
+    return '';
+  }, [conversationEntity]);
+
+  const hasCall = useMemo(() => {
+    const joinedCall = callState.joinedCall();
+    const hasEntities = conversationEntity && !!joinedCall;
+    return hasEntities && matchQualifiedIds(conversationEntity.qualifiedId, joinedCall.conversationId);
+  }, [callState, conversationEntity]);
+
+  const showCallControls = conversationEntity && ConversationFilter.showCallControls(conversationEntity, hasCall);
+
+  const isActivatedAccount = userState.isActivatedAccount();
+
+  const supportsVideoCall = conversationEntity?.supportsVideoCall(callingRepository.supportsConferenceCalling);
+
+  const isVideoCallingEnabled = teamState.isVideoCallingEnabled();
+
+  const conversationSubtitle =
+    conversationEntity.is1to1() && conversationEntity.firstUserEntity?.()?.isFederated
+      ? conversationEntity.firstUserEntity()?.handle ?? ''
+      : '';
+
+  const shortcut = Shortcut.getShortcutTooltip(ShortcutType.PEOPLE);
+  const peopleTooltip = t('tooltipConversationPeople', shortcut);
+
+  const onClickCollectionButton = () => {
+    amplify.publish(WebAppEvents.CONTENT.SWITCH, ContentViewModel.STATE.COLLECTION);
+  };
+
+  const onClickDetails = () => {
+    showConversationDetails(false);
+  };
+
+  return (
+    <ul id="conversation-title-bar" className="conversation-title-bar">
+      {conversationEntity && (
+        <>
+          <li className="conversation-title-bar-library">
+            {isActivatedAccount && (
+              <button
+                className="conversation-title-bar-icon icon-search"
```

Comment:
We are using button in a lot of places. Any reason for not having a button component in ui-kit?